feature/password-reset-and-update

WinnMoo opened the pull request:
# Issue/Ticket

CLOSES #XX
PROGRESS ON #XX

If you specify CLOSES, the ticket will be automatically closed and moved to 'done' when the PR is merged.

If you specify PROGRESS ON, a reference will be created to that issue (in both directions), but merging the PR will not close the issue.

# Description

A restart of the password reset branch due to issues with pulling from other branches

## Type of change

Feature

# How Has This Been Tested?

Unit tests and manual testing

# Notes

Put any notes here

WinnMoo committed:
Restart of the password reset branch due to issues with pulling from other branches

WinnMoo committed:
Refactored password manager according to requirements, changed email address, updated UI for password reset

alfredovargasjr commented on Backend/KFC_WebAPI/Controllers/ResetController.cs:
```diff
@@ -1,99 +1,113 @@
-                    Content = new StringContent("Email received")
-                };
-                return response;
+                string url = "localhost:8080/#/resetpassword/";
```
this is going to change once deployed

alfredovargasjr replied:
just update this now for release

WinnMoo committed:
Merge branch 'master' of https://github.com/CECS-491A/SSO into feature/reset-password-from-master

WinnMoo committed:
Updated frontend post data, fixed updating password, added pwned password scenario

julianpoy commented on Backend/DataAccessLayer/Migrations/201903051936518_PasswordReset.cs:
```diff
@@ -8,27 +8,27 @@ public partial class PasswordReset : DbMigration
+            DropIndex("dbo.PasswordResets", new[] { "UserId" });
+            DropForeignKey("dbo.PasswordResets", "UserId", "dbo.Users");
+            DropTable("dbo.PasswordResets");
         }
```
Again, this migration really can't be modified. Create a new migration that drops and re-creates this table with the correct format, or find a way to migrate the column names.

WinnMoo replied:
The problem is that I want to change the name of the table/columns in both the code and the database. 

I can't update the column names because it can't find a dbo.PasswordResetIDs table. If I change the Dbset line back to PasswordResetIDs, then the repository code that references a table called PasswordReset fails to build.

I'm stuck between two choosing two options that both won't work.

julianpoy reviewed (dismissed):
See comment regarding migrations.

WinnMoo committed:
Changed one unit test for password manager

WinnMoo committed:
Changed the body of the email that is sent when user changes their password

alfredovargasjr commented:
I fixed most of these issues on the release branch, sorry

WinnMoo committed:
Updated html tags

alfredovargasjr commented on Backend/KFC_WebAPI/Controllers/ResetController.cs:
```diff
@@ -1,99 +1,113 @@
 
 namespace WebAPI.Controllers
 {
+    [EnableCors(origins: "*", headers: "*", methods: "*")]
```
you dont need this header. The CORS config is global

alfredovargasjr replied:
Remove this

WinnMoo committed:
Updated URLs to go to kfc sso api endpoint

julianpoy commented on Backend/KFC_WebAPI/Controllers/ResetController.cs:
```diff
@@ -41,8 +41,15 @@ public IHttpActionResult SendResetEmail([FromBody] EmailRequest request)
                 PasswordManager pm = new PasswordManager();
-                string url = "localhost:8080/#/resetpassword/";
-                pm.SendResetToken(email, url);
+                string url = "https://kfc-sso.com/#/resetpassword/";
```
This should come from the incoming request. It shouldn't be hardcoded.

WinnMoo committed:
Changed url from hardcoded to get from body of request, added try/catch blocks to all reset controller methods

WinnMoo committed:
Added alerts/redirects for success password change

WinnMoo committed:
Merge branch 'master' into feature/reset-password-from-master

WinnMoo committed:
Changed variable names to follow naming conventions, removed unncessary usings in password related files

WinnMoo committed:
Removed cors tag

WinnMoo committed:
Merge branch 'master' into feature/reset-password-from-master

WinnMoo committed:
Refactored code

Deleted ExistingReset() in ResetRepository.cs
Changed the logic in ExistingReset method in ResetService.cs
Added 2 methods in PasswordManager.cs to clean up the ResetController

WinnMoo committed:
Changed how environment variables are retrieved

WinnMoo committed:
Merge branch 'master' into feature/reset-password-from-master

WinnMoo committed:
Merge branch 'master' into feature/reset-password-from-master

WinnMoo committed:
Merge branch 'master' into feature/reset-password-from-master

WinnMoo committed:
Added update password route to user dropdown menu

alfredovargasjr committed:
updated fixes and created migration for password resets

alfredovargasjr commented on Frontend/src/views/ResetPassword.vue:
```diff
@@ -0,0 +1,173 @@
+}
+</script>
+
+<style lang="css">
```
this css is overwriting other components, makes it look ugly. Make this UI match login and register, etc

alfredovargasjr commented on Frontend/src/views/SendResetLink.vue:
```diff
@@ -0,0 +1,68 @@
+</script>
+
+<style>
+.sendLink{
```
same here

alfredovargasjr commented on Frontend/src/views/UpdatePassword.vue:
```diff
@@ -0,0 +1,100 @@
+
+<style>
+
+#update{
```
same here

WinnMoo committed:
Updated UI to match login/register, refactored UserController

WinnMoo committed:
Moved requests outside of resest controller

alfredovargasjr committed:
updated navbar routes

alfredovargasjr committed:
updated home page
